src/main/java/org/prebid/server/bidder/vox/VoxBidder.java

```diff
@@ -0,0 +1,83 @@
+package org.prebid.server.bidder.vox;
+
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import io.vertx.core.http.HttpMethod;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class VoxBidder implements Bidder<BidRequest> {
+
+    private static final Map<Integer, BidType> ALLOWED_IMP_TYPES = Map.of(
+            1, BidType.banner,
+            2, BidType.video,
+            3, BidType.audio,
+            4, BidType.xNative
+    );
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public VoxBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        //todo: is it necessary to parse VoxImpExt in order to validate it?
+        final HttpRequest<BidRequest> httpRequest = HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                //todo: ORTB version header is missed, is it fine?
+                .headers(HttpUtil.headers())
+                .body(mapper.encodeToBytes(request))
+                .payload(request)
+                .build();
+        return Result.withValue(httpRequest);
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            //todo: no empty seat bids checks
+            return Result.withValues(extractBids(bidResponse));
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse("Bad Server Response"));
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    //todo: duplicate of the GothamBidder's method
```

Comment:
We had some generic logic for adapters previously, but then we understood that adapters tend to be unique(They will become with future changes).

So pls remove duplication todos.

Reply:
Also pls check similar occurences